invenio/modules/oauthclient/contrib/cern.py

```diff
@@ -0,0 +1,205 @@
+    )
+
+REMOTE_APP = dict(
+    title="CERN",
+    description="Connecting to CERN Organization.",
+    icon="",
+    authorized_handler="invenio.modules.oauthclient.handlers"
+                       ":authorized_signup_handler",
+    disconnect_handler="invenio.modules.oauthclient.handlers"
+                       ":disconnect_handler",
+    signup_handler=dict(
+        info="invenio.modules.oauthclient.contrib.cern:account_info",
+        setup="invenio.modules.oauthclient.contrib.cern:account_setup",
+        view="invenio.modules.oauthclient.handlers:signup_handler",
+    ),
+    params=dict(
+        request_token_params={
+            "resource": "datadev00.cern.ch",
+            "scope": "Name Email Bio Groups",
+        },
+        base_url="https://oauth.web.cern.ch/",
+        request_token_url=None,
+        access_token_url="https://oauth.web.cern.ch/OAuth/Token",
+        access_token_method="POST",
+        authorize_url="https://oauth.web.cern.ch/OAuth/Authorize",
+        app_key="CERN_APP_CREDENTIALS",
+        content_type="application/json",
+    )
+)
+"""CERN Remote Application."""
+
+REMOTE_SANDBOX_APP = copy.deepcopy(REMOTE_APP)
+"""CERN Sandbox Remote Application."""
+
+REMOTE_SANDBOX_APP["params"].update(dict(
+    base_url="https://test-oauth.web.cern.ch/",
+    access_token_url="https://test-oauth.web.cern.ch/OAuth/Token",
+    authorize_url="https://test-oauth.web.cern.ch/OAuth/Authorize",
+))
+
+REMOTE_APP_RESOURCE_API_URL = "https://oauthresource.web.cern.ch/api/Me"
+REMOTE_APP_RESOURCE_SCHEMA = "http://schemas.xmlsoap.org/claims"
+
+
+def fetch_groups(groups):
+    """Prepare list of allowed group names."""
+    hidden_groups = current_app.config.get(
+        'CFG_EXTERNAL_AUTH_HIDDEN_GROUPS', CFG_EXTERNAL_AUTH_HIDDEN_GROUPS)
+    hidden_groups_re = current_app.config.get(
+        'CFG_EXTERNAL_AUTH_HIDDEN_GROUPS_RE',
+        CFG_EXTERNAL_AUTH_HIDDEN_GROUPS_RE)
+    groups = [group for group in groups if group not in hidden_groups]
+    filter_groups = []
+    for regexp in hidden_groups_re:
+        for group in groups:
+            if regexp.match(group):
+                filter_groups.append(group)
+    groups = [group for group in groups if group not in filter_groups]
+
+    return groups
+
+def get_dict_from_response(response):
+    """Prepare new mapping with 'Value's groupped by 'Type'."""
+    result = {}
+    for i in response.data:
+        k = i['Type']
+        result.setdefault(k, list())
+        result[k].append(i['Value'])
+    return result
+
+
+def account_info(remote, resp):
+    """Retrieve remote account information used to find local user."""
+    # Query CERN Resources to get user info and groups
+    response = remote.get(REMOTE_APP_RESOURCE_API_URL)
+    res = get_dict_from_response(response)
+    email = res['{0}/{1}'.format(REMOTE_APP_RESOURCE_SCHEMA, 'EmailAddress')][0]
+    #nickname = res['{0}/{1}'.format(REMOTE_APP_RESOURCE_SCHEMA, 'CommonName')][0]
```

Comment:
`CommonName` should be unique in CERN database (it's my login name). If you have multiple authentication methods  enabled then it might be trickier.

I would propose `nickname=email` for now.